### DISCUSSION TITLE
Cannot install ruby 3.3.6 or 3.3.3 or any other version

### DISCUSSION THREAD

**Discussion by minhplaystudio** [rbenv/ruby-build#2476] [Build failures]
When I call rbenv install, it'll run for a while and then fail, it ask me to view the log and when I view it, this is the error:
confmk23813.tmp:48: *** missing separator.  Stop.

Sorry I'm not a Mac user, I was forced to do this. How can I solve it?

**Comment by mislav** [comment]
Hi, when you're reporting a bug like this, it really essential that you provide more information about what you tried and about your operating system, otherwise it's going to be really hard for me or anyone to guess what went wrong.

For starters, the most essential information would be: what exact command did you run, and what was its output? You can paste it into the thread like so:

~~~
```
$ rbenv install 3.3.6
<some output>
<some more output>
```
~~~

The error output of ruby-build will contain useful information like the version of ruby-build that you have, the type of your operating system, and your CPU architecture.

Then, if a log file is mentioned in the output, it's useful to attach the log file to your report. From a graphical file explorer interface, can either drag-and-drop it into a GitHub comment to attach as a file. Otherwise, you can copy/paste its contents as a GitHub Gist and link it here.

**Reply by minhplaystudio** [reply]
Sorry. I'm kinda new to this. Here are the commands I executed:

brew install rbenv
brew install ruby-build
which rbenv  (it's in /opt/homebrew/bin/rbenv)
rbenv install 3.3.6
```
ruby-build: using openssl@3 from homebrew
==> Downloading ruby-3.3.6.tar.gz...
-> curl -q -fL -o ruby-3.3.6.tar.gz https://cache.ruby-lang.org/pub/ruby/3.3/ruby-3.3.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.1M  100 21.1M    0     0  5954k      0  0:00:03  0:00:03 --:--:-- 5954k
==> Installing ruby-3.3.6...
ruby-build: using libyaml from homebrew
-> ./configure "--prefix=/usr/local/var/rbenv
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi/versions/3.3.6" --with-openssl-dir=/opt/homebrew/opt/openssl@3 --enable-shared --with-libyaml-dir=/opt/homebrew/opt/libyaml --with-ext=openssl,psych,+
-> make -j 11

BUILD FAILED (macOS 14.7.1 on arm64 using ruby-build 20241105)
```

nano /var/folders/7g/mlxhy7qx3s30k799gyr3lsgc0000gp/T/ruby-build.20241205141848.17410.log (to read the log)

```
---
-> make -j 11
Makefile:48: *** missing separator.  Stop.
external command failed with status 2
```

**Reply by minhplaystudio** [reply]
[log.log](https://github.com/user-attachments/files/18021361/log.log)
Here is the log file

**Reply by mislav** [reply]
Thanks, that helped. This line caught my eye:

```
./configure "--prefix=/usr/local/var/rbenv
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi/versions/3.3.6" ...
```

It looks really bizarre, because normally it should be all on one line and look something like:

```
./configure "--prefix=$HOME/.rbenv/versions/3.3.6" ...
```

This tells me that you probably have RBENV_ROOT environment variable set in your shell configuration files (like `.bash_profile` or `.zshrc`, and that that variable is mis-configured.

Search your shell configuration files for a line like:
```
RBENV_ROOT=/usr/local/var/rbenv\
```
Notice that there might be a backslash at the end of the line. That escapes the normal line break and causes the following bash code (`if which rbenv ...`) to be glued into the value of the variable, breaking most subsequent rbenv behavior.

You might want to consider deleting the RBENV_ROOT override entirely. The default location for rbenv data is `~/.rbenv`, which is fine for most intents and purposes. Unless you feel strongly about overriding that, keeping the defaults is usually the easiest choice.

**Comment by Adham2023** [comment]
Hello everyone,


I have the same issue, using arch linux:


```bash
<~>-> rbenv install 3.3.6
==> Downloading ruby-3.3.6.tar.gz...
-> curl -q -fL -o ruby-3.3.6.tar.gz https://cache.ruby-lang.org/pub/ruby/3.3/ruby-3.3.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.1M  100 21.1M    0     0  5433k      0  0:00:03  0:00:03 --:--:-- 5433k
==> Installing ruby-3.3.6...
-> ./configure "--prefix=$HOME/.rbenv/versions/3.3.6" --enable-shared --with-ext=openssl,psych,+
-> make -j 32

BUILD FAILED (Arch Linux  on x86_64 using ruby-build 20241213)

You can inspect the build directory at /tmp/ruby-build.20241216224553.82671.Ex5keb
See the full build log at /tmp/ruby-build.20241216224553.82671.log
<~>-1-> 

```
[ruby-build.20241216224553.82671.log](https://github.com/user-attachments/files/18154713/ruby-build.20241216224553.82671.log)

**Reply by mislav** [reply]
@Adham2023 Hi, your error seems to be completely unrelated to this thread, but let's see if we can help you out too:

The error message I'm seeing in your log is this one:
```
./tool/generic_erb.rb:6:in `require': cannot load such file -- erb (LoadError)
	from ./tool/generic_erb.rb:6:in `<main>'
make: *** [uncommon.mk:1261: encdb.h] Error 1
```

While compiling Ruby, a pre-existing version of Ruby is sometimes used. This is called "base" Ruby. From your configure log, it's this one: `BASERUBY = /usr/bin/ruby --disable=gems`. Can you check if this Ruby can load "erb"? You can run `/usr/bin/ruby -r erb -e 0` and it should exit without errors and print nothing. Otherwise, you can try installing the Arch package `ruby-erb`. (I have no idea why Arch would decide to package ruby-erb separately.)

To summarize, if an existing Ruby version is being used to help build Ruby, that existing Ruby installation has been functional with enough of a standard library present ("erb" included).

**Reply by Adham2023** [reply]
Hi, @mislav. Sorry I did not know this question is out of context. Thank you, yes it works.
